astropy/units/format/base.py

```diff
@@ -267,13 +256,15 @@ def _did_you_mean_units(cls, unit: str) -> str:
         return did_you_mean(unit, cls._units, fix=cls._fix_deprecated)
 
     @classmethod
-    def _validate_unit(cls, unit: str, detailed_exception: bool = True) -> None:
-        if unit not in cls._units:
-            if detailed_exception:
-                raise ValueError(cls._invalid_unit_error_message(unit))
-            raise ValueError()
+    def _validate_unit(cls, unit: str, detailed_exception: bool = True) -> UnitBase:
         if unit in cls._deprecated_units:
```

Comment:
Not for now, but seeing this it feels one could optimize for good units with something like,
```
try:
    return self._good_units[unit]
except KeyError:
    if unit in self._deprecated_units:
        warn
        return deprecated
    raise
```
But the test is presumably really quite quick, so no worries!

Reply:
The only formats that have any deprecated units are OGIP and VOUnit, so if they implement their own `_validate_unit()` then `_ParsingFormatMixin._validate_unit()` could safely skip the deprecated units check. And as a bonus we could get rid of the two `_deprecated_unit_warning_message()` implementations (in `_ParsingFormatMixin` and in `VOUnit`) because we could inline their code in `OGIP._validate_unit()` and `VOUnit._validate_unit()`.